Backend/app/provider-platform/dynamic-offer-driver-app/Main/spec/Storage/WhereIsMyBus.yaml

```diff
@@ -70,25 +82,62 @@ TripTransaction:
     endLocation: Storage.Queries.Transformers.Ride.mkLatLong|I
   domainInstance:
     - Custom Kernel.Beam.Lib.UtilsTH.mkBeamInstancesForEnumAndList <TripStatus>
-DriverRequest:
-  tableName: driver_request
+  queries:
+    findByTransactionId:
+      kvFunction: findOneWithKV
+      where: id
+    updateOnStart:
+      kvFunction: updateOneWithKV
+      params: [routeCode, startLocation, status]
+      where: id
+    findAllTripTransactionByDriverId:
+      kvFunction: findAllWithKV
+      where: driverId
+    updateStatus:
+      kvFunction: updateOneWithKV
+      params: [status, endLocation]
+      where: id
+
+  extraOperations:
+    - EXTRA_QUERY_FILE
+
+ApprovalRequest:
+  tableName: approval_request
 
   types:
     RequestStatus:
-      enum: "ACCEPTED, REJECTED"
+      enum: "ACCEPTED, REJECTED, PENDING"
+
+    ApprovalRequestData:
+      enum: "WmbEndTrip WmbEndTripData"
+
+    WmbEndTripData:
+      lat: Double
+      lon: Double
+      reason: Maybe Text
+      tripTransactionId: Id TripTransaction
+      derive: "Eq,Ord,Read"
 
   fields:
-    id: Id DriverRequest
-    requestType: Text
-    description: Maybe Text
-    tripTransactionId : Id TripTransaction
```

Comment:
without tripTransactionId, how will we link which request is approved for which driver and which trip ?